### PR TITLE
fix_duplication_of_outputs_bug

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

**Comment by fluidvanadium** [comment]
branch name implies a bug fix... but i cant identify the bug

File: zingolib/src/wallet/transaction_record.rs [hunk 1]
```diff
@@ -185,30 +185,13 @@ impl TransactionRecord {
...
-            saplings = notes::AnyPoolOutput::get_all_outputs_with_status(self, spend_status_query);
-        }
-        if *include_notes.orchard() {
-            orchards = notes::AnyPoolOutput::get_all_outputs_with_status(self, spend_status_query);
-        }
-        set.extend(transparents);
-        set.extend(saplings);
-        set.extend(orchards);
-        set
+        notes::AnyPoolOutput::get_all_outputs_with_status(self, *include_notes.spend_status())
```
Comment (fluidvanadium) [inline zingolib/src/wallet/transaction_record.rs]: half of the output query is dropped here

**Review by fluidvanadium — DISMISSED** [review]
im doing some cleanup on top of this RN

File: zingolib/src/wallet/transaction_record.rs (hunk 1 above)
Comment (Oscar-Pepper) [reply zingolib/src/wallet/transaction_record.rs]: this is true, this fn takes an OutputQuery including a requested pool but this removes the pool logic

**Review by Oscar-Pepper — DISMISSED** [review]

**Comment by zancas** [comment]
@Oscar-Pepper @fluidvanadium updated and fixed.

File: zingolib/src/wallet/notes.rs [hunk 2]
```diff
@@ -79,6 +80,11 @@ impl AnyPoolOutput {
             )
             .collect()
     }
+    /// Uses a query to select all notes with specific properties and returns
+    /// a vector packing them in the AnyPoolOutput
+    pub fn get_all_outputs(query: &TransactionRecord) -> Vec<Self> {
+        Self::get_outputs_by_status(query, OutputSpendStatusQuery::any())
+    }
```
Comment (fluidvanadium) [inline zingolib/src/wallet/notes.rs]: this function is confused. its parameter is a TransactionRecord named query and though the comment describes a filter, there is none

**Review by fluidvanadium — COMMENTED** [review]
i echoed this change in #1243. list_outputs calls get_outputs for each TR and flat maps them together

File: zingolib/src/wallet/notes.rs (hunk 2 above)
Comment (zancas) [reply zingolib/src/wallet/notes.rs]: Addressing...

**Review by fluidvanadium — APPROVED** [review]

**Pull request merged**